src/main/java/com/sonymobile/jenkins/plugins/mq/mqnotifier/QueueListenerImpl.java

```diff
@@ -61,6 +62,10 @@ public void onLeft(Queue.LeftItem li) {
             json.put(Util.KEY_DEQUEUE_REASON, Util.VALUE_CANCELLED);
         } else {
             json.put(Util.KEY_DEQUEUE_REASON, Util.VALUE_BUILDING);
+            Label assignedLabel = li.getAssignedLabel();
+            json.put(Util.KEY_DEQUEUE_ALLOCATED_LABEL,
+                assignedLabel != null ? assignedLabel.getDisplayName() : Util.KEY_DEQUEUE_NO_LABEL);
```

Comment:
Would it be better to use a null value or omit the key completely rather than pick a random string to represent the lack of a label?